src/super_gradients/training/transforms/keypoints/keypoints_mosaic.py

```diff
@@ -71,18 +73,27 @@ def __init__(self, prob: float, pad_value=(127, 127, 127)):
         :param prob:     Probability to apply the transform.
         :param pad_value Value to pad the image if size of samples does not match.
         """
-        super().__init__(additional_samples_count=3)
+        super().__init__()
         self.prob = prob
         self.pad_value = tuple(pad_value)
 
+    @property
+    def additional_samples_count(self) -> int:
+        do_mosaic = random.random() < self.prob
+        return 3 if do_mosaic else 0
+
+    @additional_samples_count.setter
+    def additional_samples_count(self, value):
+        pass  # ignored
```

Comment:
Maybe let's throw an exception to explicitly disable writing to this property?

Reply:
Done, [49f5548](https://github.com/Deci-AI/super-gradients/pull/1897/commits/49f55484f022f5938ef472a5eed5900a26fe9d2b)